### PR TITLE
[QC-932] add a new repo cleaner policy

### PR DESCRIPTION
**Pull request by Barthelemy**
because the production policy was too complicated and this is good enough according to detectors.

### CODE REVIEW

File: Framework/script/RepoCleaner/qcrepocleaner/rules/multiple_per_run.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    update_list: List[ObjectVersion] = []
+    versions_buckets_dict: DefaultDict[str, List[ObjectVersion]] = defaultdict(list)
+    metadata_for_preservation = {'preservation': 'true'}
+
+    # config parameters
+    period_pass = (extra_params.get("period_pass", False) == True)
+    logger.debug(f"period_pass : {period_pass}")
+    interval_between_versions = int(extra_params.get("interval_between_versions", 30))
+    logger.debug(f"interval_between_versions : {interval_between_versions}")
+    migrate_to_EOS = (extra_params.get("migrate_to_EOS", False) is True)
```
Comment (knopers8) [inline Framework/script/RepoCleaner/qcrepocleaner/rules/multiple_per_run.py:62]: Is there a reason why you compare `period_pass` using `== True`, but `migrate_to_EOS` with `is True`?

File: Framework/script/RepoCleaner/qcrepocleaner/rules/multiple_per_run.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+def process(ccdb: Ccdb, object_path: str, delay: int,  from_timestamp: int, to_timestamp: int,
+            extra_params: Dict[str, str]):
+    '''
+    Process this deletion rule on the object. We use the CCDB passed by argument.
+    Objects who have been created recently are spared (delay is expressed in minutes).
+
+    This specific policy, multiple_per_run, operates like this : Keep the first and the last version of a run plus
+    1 version every X minutes.
+
+    Extra parameters:
```
Comment (knopers8) [inline Framework/script/RepoCleaner/qcrepocleaner/rules/multiple_per_run.py:23]: `period_pass` and `object_attribute` are not explained as extra parameters

**Review by knopers8 — COMMENTED** [review]
Some minor questions, but it looks good overall, thank you

File: Framework/script/RepoCleaner/qcrepocleaner/policies_utils.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+from datetime import timedelta
+import logging
+from typing import DefaultDict, List
+
+from qcrepocleaner.Ccdb import Ccdb, ObjectVersion
+
+logger = logging  # default logger
+
+def in_grace_period(version: ObjectVersion, delay: int):
+    return not (version.validFromAsDt < datetime.now() - timedelta(minutes=delay))
```
[the remark this answers is not in the document]
Comment (Barthelemy) [reply Framework/script/RepoCleaner/qcrepocleaner/policies_utils.py:11]: indeed, we should perhaps move to createdAt here. I have already used it in the new policy.

File: Framework/script/RepoCleaner/qcrepocleaner/rules/multiple_per_run.py (hunk 1 above)
Comment (Barthelemy) [reply Framework/script/RepoCleaner/qcrepocleaner/rules/multiple_per_run.py:62]: good point. Modern Python dictates to use `is True`. I will change the other one.

File: Framework/script/RepoCleaner/qcrepocleaner/rules/multiple_per_run.py (hunk 2 above)
Comment (Barthelemy) [reply Framework/script/RepoCleaner/qcrepocleaner/rules/multiple_per_run.py:23]: yes. and actually `object_attribute` is not used anymore.

**Review by knopers8 — APPROVED** [review]
thank you

**Pull request merged**